pkg/pillar/cmd/diag/diag.go

```diff
@@ -881,6 +878,18 @@ func tryLookupIP(ctx *diagContext, ifname string) bool {
 func tryPing(ctx *diagContext, ifname string, reqURL string) bool {
 
 	zedcloudCtx := ctx.zedcloudCtx
+	if zedcloudCtx.TlsConfig == nil {
+		err := zedcloud.UpdateTLSConfig(zedcloudCtx, ctx.serverName, ctx.cert)
+		if err != nil {
+			log.Errorf("internal UpdateTLSConfig failed %v\n", err)
+			return false
+		}
+		if zedcloudCtx.TlsConfig == nil {
```

Comment:
Do we need this check? Is checking for error above not enough?

Reply:
this check is for the TLS resumption, so it only gets from the tlsConfig once, then keep using the same one. but only if the proxy cert is changed, that stack above shown will update the tlsconfig setting

Reply:
I mean we [return](https://github.com/lf-edge/eve/blob/3d38ec804d55051d83e24dcd374709fdb73a7d06/pkg/pillar/zedcloud/tls.go#L181) nil error only when we [set](https://github.com/lf-edge/eve/blob/3d38ec804d55051d83e24dcd374709fdb73a7d06/pkg/pillar/zedcloud/tls.go#L62) TlsConfig.

Reply:
Ok, make sense, removed.